sklearn/gaussian_process/_gpr.py

```diff
@@ -198,7 +198,9 @@ def fit(self, X, y):
             self._y_train_std = np.std(y, axis=0)
 
             # Remove mean and make unit variance
-            y = (y - self._y_train_mean) / self._y_train_std
+            # Moreover, add a very small number to the y_train.std to
+            # avoid a divide by zero error.
+            y = (y - self._y_train_mean) / (self._y_train_std + 1E-19)
```

Comment:
Line 203 will always add an epsilon even when `_y_train_std` is not 0.
Do you mind adding an if check as this will be done only when `_y_train_std==0`?